### PR TITLE
User management

### PR DESCRIPTION
**Pull request by p1003**

### CODE REVIEW

**Comment by grzanka** [comment]
Lets imagine I have my containers up and running , according to `SHIELDHIT_PATH=path/to/shieldhit docker-compose up -d --build` instruction.
I modify the `yaptide/data/script_input.json` file, adding some new instruction.

We need some section in main `README.md` explaining clearly how to run it. Should it be executed inside docker ?
If executed outside docker, then what requirements should be installed ?
When running outside docker with `pip install -r requirements.txt` the scripts complains:

```
gitpod /workspace/yaptide/yaptide/data $ python db_script.py 
Traceback (most recent call last):
  File "/workspace/yaptide/yaptide/data/db_script.py", line 63, in <module>
    insert_user(con=connection, data=obj["DATA"])
  File "/workspace/yaptide/yaptide/data/db_script.py", line 24, in insert_user
    users = db.Table('User', metadata, autoload=True, autoload_with=engine)
  File "<string>", line 2, in __new__
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/util/deprecations.py", line 309, in warned
    return fn(*args, **kwargs)
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 616, in __new__
    metadata._remove_table(name, schema)
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
    raise exception
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 611, in __new__
    table._init(name, metadata, *args, **kw)
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 686, in _init
    self._autoload(
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 721, in _autoload
    conn_insp.reflect_table(
  File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/engine/reflection.py", line 789, in reflect_table
    raise exc.NoSuchTableError(table_name)
sqlalchemy.exc.NoSuchTableError: User
```

File: yaptide/data/db_script.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+if __name__ == "__main__":
+    input_json_path = Path(os.path.dirname(os.path.realpath(__file__)), 'script_input.json')
+    with open(input_json_path) as json_file:
+        json_data = json.load(json_file)
+    engine = db.create_engine('sqlite:///main.db')
+    connection = engine.connect()
+
+    for obj in json_data:
+        if obj["OPERATION"] == "INSERT" and obj["TABLE"] == "User":
```
Comment (grzanka) [inline yaptide/data/db_script.py]: Put "OPERATION", "TABLE" and other hardcoded strings into a single (even global) variables.

**Review by grzanka — DISMISSED** [review]
See comments

**Comment by p1003** [comment]
> Lets imagine I have my containers up and running , according to `SHIELDHIT_PATH=path/to/shieldhit docker-compose up -d --build` instruction. I modify the `yaptide/data/script_input.json` file, adding some new instruction.
> 
> We need some section in main `README.md` explaining clearly how to run it. Should it be executed inside docker ? If executed outside docker, then what requirements should be installed ? When running outside docker with `pip install -r requirements.txt` the scripts complains:
> 
> ```
> gitpod /workspace/yaptide/yaptide/data $ python db_script.py 
> Traceback (most recent call last):
>   File "/workspace/yaptide/yaptide/data/db_script.py", line 63, in <module>
>     insert_user(con=connection, data=obj["DATA"])
>   File "/workspace/yaptide/yaptide/data/db_script.py", line 24, in insert_user
>     users = db.Table('User', metadata, autoload=True, autoload_with=engine)
>   File "<string>", line 2, in __new__
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/util/deprecations.py", line 309, in warned
>     return fn(*args, **kwargs)
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 616, in __new__
>     metadata._remove_table(name, schema)
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
>     compat.raise_(
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
>     raise exception
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 611, in __new__
>     table._init(name, metadata, *args, **kw)
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 686, in _init
>     self._autoload(
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 721, in _autoload
>     conn_insp.reflect_table(
>   File "/home/gitpod/.pyenv/versions/3.9.6/lib/python3.9/site-packages/sqlalchemy/engine/reflection.py", line 789, in reflect_table
>     raise exc.NoSuchTableError(table_name)
> sqlalchemy.exc.NoSuchTableError: User
> ```

Currently it has to be run from folder in which it is saved because it has relative path to `main.db` file. I can add reading absolute path. 

There are also no other requirements than those needed for flask server to run.

**Comment by grzanka** [comment]
If it needs to run in a certain folder, then add an appropriate section in the documentation. Also - be specific if the command needs to be executed inside or outside docker container.

**Comment by p1003** [comment]
> If it needs to run in a certain folder, then add an appropriate section in the documentation. Also - be specific if the command needs to be executed inside or outside docker container.

When trying to run `db_script.py` from outside of docker you get error: `sqlite3.OperationalError: database is locked`. It is due to fact that docker actively uses database and database can handle only 1 process at a time. So `db_script.py` has to run from inside docker container but now it can be run from any folder

File: README.md [hunk 2]
```diff
@@ -92,6 +92,10 @@ curl -i -X GET -b cookies.txt -c cookies.txt http://localhost:5000/auth/refresh
 curl -i -X DELETE http://localhost:5000/auth/logout
 ```
 
+# Docker database
+
+Now registering, updating and deleting users is available with the use of ``db_script.py`` located in ``yaptide/data`` folder, which is a volume folder for ``yaptide_flask`` container. To make any changes in the database, ``script_input.json`` (the name is currently hardcoded) file should be prepared. ``example_script_input.json`` shows pattern of doing this. It is possible to either edit the file in Docker CLI (user unfriendly way) or prepare the file elsewhere and copy/move it into ``yaptide_data/_data`` folder found in docker volumes folder. After preparing the file, ``db_script.py`` can be run in the Docker CLI (from any folder) of ``yaptide_flask`` container. It is not possible to run it outside of the docker because of the database concurrency limitations.
```
Comment (grzanka) [inline README.md]: Add here copy-pasteable instruction how to run the `db_script.py` inside already running docker container, assuming the containers were started using docker compose as described above.
I assume the script can be run inside previously started container, not only during startup.

**Review by p1003 — COMMENTED** [review]
LGBT
Waiting for your approval @grzanka

**Review by grzanka — APPROVED** [review]
LGTM

**Pull request merged**